### ISSUE TITLE
[Feature Request] Improve usability of programmatic API

### ISSUE THREAD

**Issue by apsdehal** [VikParuchuri/surya#16]
Currently, the way the programmatic API is setup, it's not easy to customize it. Two examples that I have seen so far include:

- `dpi` parameter is not exposed to the `load_pdf` util and `get_page_images` is not [leveraging](https://github.com/VikParuchuri/surya/blob/c2067a21b857669cb897623a8643bbeaedc6dc80/surya/model/processing.py#L50) the DPI set in the settings.
- Since the settings object is created at the import time and `get_detected_boxes` function [here](https://github.com/VikParuchuri/surya/blob/c2067a21b857669cb897623a8643bbeaedc6dc80/surya/postprocessing/heatmap.py#L99) uses the settings object directly in the function definition, it's not possible to modify these settings when using surya programmatically. 

Happy to open up a PR to fix these and any future ones that I see if there is an interest in accepting PR contributions.

**Comment by shividhar** [comment]
Agreed, this would be very useful. The documentation suggests these settings can be overridden with envrionment variables, but additional instructions would be great: `Inspect the settings in surya/settings.py. You can override any settings with environment variables.` ([link](https://github.com/VikParuchuri/surya?tab=readme-ov-file#usage))

**Comment by Shu-Wan** [comment]
Is is possible to add a [prefix](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#environment-variable-names) to all environment variables?

Some of the variable names are pretty generic, so adding a prefix would help avoid accidental conflicts.




```python
class Settings(BaseSettings):
    model_config = SettingsConfigDict(env_prefix="my_prefix_")
```
